src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs

```diff
@@ -11,6 +12,1496 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public partial class IOperationTests : SemanticModelTestBase
     {
+        private static readonly CSharpParseOptions ImplicitObjectCreationOptions = TestOptions.RegularPreview;
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void ImplicitObjectCreationWithArguments()
+        {
+            string source = @"
+class C
+{
+    public C(byte i, long j)
+    {
+    }
+    void M()
+    /*<bind>*/{
+        C x1 = new(3, 4);
+        C x2 = new(j: 3, i: 4);
+        C x3 = new(3, j: 4);
+        C x4 = new(i: 3, 4);
+    }/*</bind>*/
+}
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            string expectedOperationTree = @"
+IBlockOperation (4 statements, 4 locals) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+  Locals: Local_1: C x1
+    Local_2: C x2
+    Local_3: C x3
+    Local_4: C x4
+  IVariableDeclarationGroupOperation (1 declarations) (OperationKind.VariableDeclarationGroup, Type: null) (Syntax: 'C x1 = new(3, 4);')
+    IVariableDeclarationOperation (1 declarators) (OperationKind.VariableDeclaration, Type: null) (Syntax: 'C x1 = new(3, 4)')
+      Declarators:
+          IVariableDeclaratorOperation (Symbol: C x1) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'x1 = new(3, 4)')
+            Initializer: 
+              IVariableInitializerOperation (OperationKind.VariableInitializer, Type: null) (Syntax: '= new(3, 4)')
+                IObjectCreationOperation (Constructor: C..ctor(System.Byte i, System.Int64 j)) (OperationKind.ObjectCreation, Type: C) (Syntax: 'new(3, 4)')
```

Comment:
It feels like the IDE or an analyzer will need to know at some point whether the object creation was target-typed or not. We should add a flag to IObjectCreationOperation, and conditionally include it in the writer when true.

Reply:
Adding such field to BoundObjectCreation (to use in IOperation) would require to update all use-sites. Unfortunately we can't use WasConverted property since it's debug-only.

Reply:
What approach do we want to take here? /cc @gafter

Reply:
It shouldn't require updating all use-sites. Add the field to the node, add a constructor overload in `Constructors.cs` for the original node that specifies it's not inferred, and use the new constructor when creating a target-typed constructor.

Reply:
we reuse the existing BindObjectCreation and others to bind `new()`. I guess there's no way around it, I'll add it to required code paths.